app/failed_logins/rest.py

```diff
@@ -0,0 +1,84 @@
+from flask import Blueprint, jsonify
+
+from app.dao.failed_logins_dao import (
+    dao_create_failed_login_for_ip,
+    dao_get_count_of_all_failed_logins_for_ip,
+    dao_get_failed_logins,
+    dao_get_latest_failed_login_by_ip,
+)
+from app.errors import InvalidRequest, register_errors
+from app.utils import (
+    calculate_delay_period,
+    check_request_within_throttle_period,
+    get_ip_address,
+    log_throttled_login,
+)
+
+failed_logins_blueprint = Blueprint(
+    "failed_logins",
+    __name__,
+    url_prefix="/failed-logins",
+)
+
+register_errors(failed_logins_blueprint)
+
+
+@failed_logins_blueprint.route("fetch-all")
+def get_all_failed_logins():
+    """
+    Fetches all failed login records in the table.
+    """
+    data = dao_get_failed_logins()
+    data = [obj.serialize() for obj in data] if data else []
+    return jsonify(data or {}), 200
+
+
+@failed_logins_blueprint.route("fetch-for-requester")
+def get_failed_login_for_requester():
+    """
+    Fetches most recent failed login attempt for specific IP address from the table.
+    """
+    ip = get_ip_address()
+    if not ip:
+        errors = {"ip": ["Missing data for required field."]}
+        raise InvalidRequest(errors, status_code=400)
+    data = dao_get_latest_failed_login_by_ip(ip)
+    return jsonify(data.serialize() if data else {}), 200
+
+
+@failed_logins_blueprint.route("check-failed-login-for-requester")
+def check_throttle_for_requester(user):
```

Comment:
This function doesn't use the user variable, so you probably shouldn't bother passing it in